Configurations/unix-Makefile.tmpl

```diff
@@ -1484,8 +1484,11 @@ EOF
                  grep { platform->isdef($_) }
                  @{$args{objs}};
       my @deps = compute_lib_depends(@{$args{deps}});
-      my $shared_def = join("", map { ' '.$target{shared_defflag}.$_ } @defs);
-
+      if (m/providers\/fips/ && defined $target{shared_fipsflag}) {
+        $shared_def = join("", map { ' '.$target{shared_fipsflag}.$_ } @defs);
+      } else {
+        $shared_def = join("", map { ' '.$target{shared_defflag}.$_ } @defs);
+      }
```

Comment:
The reason I ask about the order of what you currently have in `$target{shared_fipsflag}` is that you could easily make it an *additional* flag instead of a replacement.  In other words, you could have this in 10-main.conf:

``` perl
          shared_fipsflag  => "-Wl,-binitfini:init:cleanup",
```

... and then just this here:

``` diff
       my $shared_def = join("", map { ' '.$target{shared_defflag}.$_ } @defs);
+      $shared_def .= ' '.$target{shared_fipsflag} if defined($target{shared_fipsflag});
```

Reply:
I see. Yes, this seems more elegant. Will change the PR tomorrow.

Reply:
Hm. Wait, it will be more like ...
```diff
+      $shared_def .= ' '.$target{shared_fipsflag} if (m/providers\/fips/ && defined($target{shared_fipsflag}));
```
As this flag's content must only be used for the fips provider.

Reply:
Yeah, that kinda works, although I prefer to avoid "special knowledge" like that in the build file templates.  If you don't mind, I can add a refactoring commit to this PR after you're done?

Reply:
good work @makr 
Thanks for looking into this.